compiler/noirc_evaluator/src/ssa/opt/constant_folding/mod.rs

```diff
@@ -276,14 +317,25 @@ impl Context {
                     }
 
                     self.values_to_replace.batch_insert(&old_results, cached);
+
                     return;
                 }
-                CacheResult::NeedToHoistToCommonBlock(dominator) => {
+                CacheResult::NeedToHoistToCommonBlock { dominator, .. } if dominator == block => {
+                    // We found an instruction to cached in an origin that this block dominates.
```

Comment:
Can you clarify this comment? Can we basically find a common dominator that is the block itself and thus we want to avoid revisiting it causing a cycle? Let's note this reason if it is the case.

Reply:
I'll investigate again how this happens and make it clearer. I thought this won't happen with our order of visiting blocks, now that we don't revisit them, but then it did. I thought maybe it was a loop back edge. Or maybe it's outdated, will check.

Reply:
I think this is also something that only happened when revisits retained the cache. Changed it into an assertion that it does not happen.